Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/CONNECTSamlAssertionValidator.java

```diff
@@ -120,22 +120,35 @@ private static void validateIssuer(Issuer issuer) throws WSSecurityException {
         if (SamlConstants.URN_OASIS_NAMES.equals(issuer.getFormat())) {
             if (StringUtils.isNotBlank(issuer.getSPProvidedID())) {
                 throw new WSSecurityException(ErrorCode.FAILED_CHECK,
-                    SamlConstants.SECURITY_ASSERTION_ISSUER_FORMAT + " " + SamlConstants.URN_OASIS_NAMES + " " + "and"
-                        + SamlConstants.SECURITY_ASSERTION_ISSUER + " " + IS_PRESENT);
+                    getCustomErrorMsg(SamlConstants.SECURITY_ASSERTION_SPPROVIDEDID));
             }
             if (StringUtils.isNotBlank(issuer.getNameQualifier())) {
                 throw new WSSecurityException(ErrorCode.FAILED_CHECK,
-                    SamlConstants.SECURITY_ASSERTION_ISSUER_FORMAT + " " + SamlConstants.URN_OASIS_NAMES + " " + "and"
-                        + SamlConstants.SECURITY_ASSERTION_ISSUER + " " + IS_PRESENT);
+                    getCustomErrorMsg(SamlConstants.SECURITY_ASSERTION_NAME_QUALIFIER));
             }
-
             if (StringUtils.isNotBlank(issuer.getSPNameQualifier())) {
                 throw new WSSecurityException(ErrorCode.FAILED_CHECK,
-                    SamlConstants.SECURITY_ASSERTION_ISSUER_FORMAT + " " + SamlConstants.URN_OASIS_NAMES + " " + "and"
-                        + SamlConstants.SECURITY_ASSERTION_ISSUER + " " + IS_PRESENT);
-
+                    getCustomErrorMsg(SamlConstants.SECURITY_ASSERTION_SPNAME_QUALIFIER));
             }
         }
+    }
+
+    /**
+     * Generate custom error msg based on template for validating assertion issuer
+     *
+     * @param errorType error type
+     * @return custom error msg
+     */
+    private static String getCustomErrorMsg(final String errorType) {
+        StringBuilder errorBuilder = new StringBuilder();
+        errorBuilder.append(SamlConstants.SECURITY_ASSERTION_ISSUER_FORMAT);
+        errorBuilder.append(" ");
+        errorBuilder.append(SamlConstants.URN_OASIS_NAMES);
+        errorBuilder.append(" and ");
+        errorBuilder.append(errorType);
+        errorBuilder.append(" ");
+        errorBuilder.append(IS_PRESENT);
+        return errorBuilder.toString();
```

Comment:
Suggestion for readability -- chaining calls to append:

```java
StringBuilder errorBuilder = new StringBuilder(SamlConstants.SECURITY_ASSERTION_ISSUER_FORMAT)
    .append(" ")
    ...SNIPPED...
```

If you're feeling really crazy, you can even skip the StringBuilder declaration and just make the function a single statement:

```java
return new StringBuilder(SamlConstants.SECURITY_ASSERTION_ISSUER_FORMAT)
    .append(" ")
    ...SNIPPED...
    .toString();
```